enclave/certificate_gen.go

```diff
@@ -0,0 +1,85 @@
+package enclave
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
```

Comment:
It's cool that this is part of the standard library. We may want to split it out into a separate project or binary called crux-utils or similar. But I'd be interested to get your views.